runtime/src/migrations/nft_multi.rs

```
@@ -0,0 +1,683 @@
+	pub struct TokenOwnership<AccountId, MaxTokensPerCollection>
+	where
+		AccountId: Debug + PartialEq + Clone,
+		MaxTokensPerCollection: Get<u32>,
+	{
+		pub owner: AccountId,
+		pub owned_serials: BoundedVec<SerialNumber, MaxTokensPerCollection>,
+	}
+}
+
+/// Convert from old CollectionInfo type to new type
+/// Returns a bool if the CollectionInfo has been completely migrated, if not, this function
+/// will re-insert the old CollectionInfo into the storage
+fn convert<T: pallet_nft::Config>(
+	collection_id: CollectionUuid,
+	old: &mut old::CollectionInformation<T::AccountId, T::MaxTokensPerCollection, T::StringLimit>,
+) -> (bool, TokenCount) {
+	log::debug!(target: LOG_TARGET, "🦆 Migrating collection_id {}", collection_id);
+	// Is the collection info fully migrated
+	let mut completed = false;
+	let mut migrated_token_count: TokenCount = 0;
+
+	// For simplicity, we will only migrate max 50 tokens from 1 account at a time
+	if let Some(mut ownership) = old.owned_tokens.pop() {
+		let mut serial_numbers = ownership.owned_serials.clone();
+		// take at max 10 from serial_numbers
+		migrated_token_count = serial_numbers.len().min(MAX_TOKENS_PER_STEP as usize) as TokenCount;
+		let serials_to_migrate = serial_numbers.drain(..migrated_token_count as usize).collect();
+		log::debug!(target: LOG_TARGET, "🦆 Migrating {:?} tokens for owner: {:?}", serials_to_migrate, ownership.owner);
+
+		for serial_number in &serials_to_migrate {
+			let token_locks = old::TokenLocks::<T>::take((collection_id, serial_number));
+			let token_utility_flags =
+				old::TokenUtilityFlags::<T>::take((collection_id, serial_number));
+
+			let token_info = TokenInformation {
+				owner: ownership.owner.clone(),
+				lock_status: token_locks,
+				utility_flags: token_utility_flags,
+			};
+			pallet_nft::TokenInfo::<T>::insert(collection_id, serial_number, token_info);
+		}
+
+		// Update OwnedTokens with the migrated serials
+		let _ = pallet_nft::OwnedTokens::<T>::try_mutate(
```

we ignored the error in owned tokens update

if let Err(e) = pallet_nft::OwnedTokens::<T>::try_mutate(
    &ownership.owner,
    collection_id,
    |owned_serials| -> DispatchResult {
        // ...
        Ok(())
    },
) {
    log::error!(target: LOG_TARGET, "🦆 Error updating owned tokens: {:?}", e);
    // Consider how to handle this error - maybe mark the migration as failed?
}

---

The try_mutate will never error as we always return Ok(())
In both places where it could have errored, we are using force_push and truncate_from which do not return a Result. These are both safe as the BoundedVec upper bound will not change between the old and new datasets
so it is safe to ignore the result altogether